### ISSUE TITLE
How to enable ape?

### ISSUE THREAD

**Issue by md5xwl** [open-mmlab/mmclassification#890]
I want to train EfficientNet. it need large cuda memery.  APE can reduce memory requirements.

it is easy to enable ape in pytorch:
        with autocast():
            output = model(images)
            loss = criterion(output, target)
        scaler.scale(loss).backward()
        scaler.step(optimizer)
        scaler.update()

so Where should I add it.

**Comment by mzr1996** [comment]
If you want to enable amp training, just add `fp16 = dict(loss_scale='dynamic')` in your config.
Otherwise, if you want to modify the training procedure by yourself, maybe you need to dive into mmcv, for example, the model forward is called in [EpochBasedRunner.run_iter](https://github.com/open-mmlab/mmcv/blob/d1004b7c92ab472b2d69bafc126ede4478031cf6/mmcv/runner/epoch_based_runner.py#L31), and the optimizer backward is call in [OptimizerHook.after_train_iter](https://github.com/open-mmlab/mmcv/blob/d1004b7c92ab472b2d69bafc126ede4478031cf6/mmcv/runner/hooks/optimizer.py#L61)
